vmaas/rpm_pkg_names.go

```diff
@@ -0,0 +1,53 @@
+package vmaas
+
+import (
+	"slices"
+
+	"github.com/pkg/errors"
+	"github.com/redhatinsights/vmaas-lib/vmaas/utils"
+)
+
+type ContentData map[string][]string
+
+type RPMPkgNames struct {
+	Names      ContentData `json:"rpm_name_list"`
+	LastChange string      `json:"last_change"`
+}
+
+func (c *Cache) getContentData(names []string, contentSets []string) ContentData {
+	csMap := make(map[string]bool, len(contentSets))
+	for _, cs := range contentSets {
+		csMap[cs] = true
+	}
+
+	isDuplicate := make(map[string]bool, len(names))
+	contentData := make(ContentData, len(names))
+	for _, name := range names {
+		nameID, found := c.Packagename2ID[name]
+		if !found || isDuplicate[name] {
+			continue
+		}
+
+		csIDs := c.PkgNameID2ContentSetIDs[nameID]
```

Comment:
please stick to the original implementation with `ContentSetID2PkgNameIDs` and remove `PkgNameID2ContentSetIDs` from cache. We should try to keep the cache minimal and since there should be less content sets than package names we should use contentSet2packagename mapping. Extra iteration through slice of package names from this map should be fast enough for this API.